docs/RFCS/beta_and_experimental_features.md

```diff
@@ -0,0 +1,220 @@
+*expecting* the feature to *remain available* in the future. Since
+CockroachDB also advertises backward compatibility across releases,
+this expectation seems reasonable at first.
+
+Yet, when new features are added to CockroachDB, there is a period of
+time where the UX aspects of the feature may not be stabilized yet: we
+may not be certain yet that the UX choices are good, or perhaps early
+users will provide feedback that change our UX choices. Thus, the
+CockroachDB team may want to preserve the ability to change the UX
+aspects without violating the promise of backward compatibility.
+
+This is analogous to the process of "API stabilization" when a library
+or online services adds a new feature -- the feature is initially
+released, then early adopters will try it out, and then based on the
+feedback the API may change slightly before it is stabilized and
+becomes "frozen" for the purpose of defining backward compatibility.
+In this initial phase, the new feature/API is typically marked
+"experimental" in its documentation.
+
+In contrast to libraries or online services, however, the "API" of
+CockroachDB has a large surface area, and there is not a single point
+in the product nor the documentation where "features" are identified
+as a bullet list, where it would be convenient to add the text
+"experimental" next to their description.
+
+More to the point, a SQL database is typically a product where users
+are likely to discover features via experimentation as opposed to
+checking the documentation beforehand. Without a clear mention *within
+the product* a users may fail to learn altogether that a feature is
+experimental, and erroneously start to develop an expectation of
+backward compatibility.
+
+This motivates guideliens to make the experimental nature of new
+functionality more clear throughout the product, not only the
+documentation.
+
+# Guide-level explanation
+
+When implementing new features first determine whether the feature
+should be considered beta/experimental at all. For this refer to the
+section below.
+
+Then, if applicable, to display the experimental nature of the feature
+to the UX elements explicitly:
+
+- for the log messages: TBD?
+- for command-line flags: add the marker `[EXPERIMENTAL]` in the help text (`--help`);
+- for cluster settings: add the marker `[EXPERIMENTAL]` in the setting description (column "Description" in `SHOW ALL CLUSTER SETTINGS`).
+- for session variables: add the marker `[EXPERIMENTAL]` in the variable description (column "Description" in `SHOW ALL`, also see below).
+- for local commands in the CLI shell: add the marker `[EXPERIMENTAL]` in the online help (output of `\h`)
+- for environment variables: add the word `EXPERIMENTAL` in the name of the environment variable.
+- for SQL contextual help: add the word `[EXPERIMENTAL]` in the description.
+- for the online documentation: be explicit that the feature is experimental.
```

Comment:
We can find a consistent way to do this, perhaps including `Experimental` in the page title, etc. We should probably also add a page explaining what this status means and listing all current experimental features.